### PR TITLE
DQN tutorial fixes

### PR DESCRIPTION
**Pull request by SiftingSands**
Gym's domain name has changed since https://github.com/pytorch/tutorials/pull/1928#issue-1247029039

Minor updates to Gym calls, which would produce errors with the latest version of Gym (0.25.2). (possibly fixes https://github.com/pytorch/tutorials/issues/1432#issue-837212016)

Still doesn't train properly as noted in https://github.com/pytorch/tutorials/issues/1755#issue-1071634106 , but at least it runs now as shown in the figure below (didn't touch any other settings in the tutorial).

Figure_2

### CODE REVIEW

**Review by vmoens — APPROVED** [review]
Thanks @SiftingSands for this
LGTM, if you don't mind I'll have a deeper look at the tuto and check why it doesn't learn before merging.
Forgive me for putting this on hold!

**Comment by SiftingSands** [comment]
@vmoens No worries at all. I did try tweaking various hyperparameters and doing reward shaping to get it to train within 500 to 1000 episodes. As with others, I saw non-monotonic reward history, where the agent would perform well then forget how to solve the problem again. I believe this is to be expected for vanilla DQN learning from pixels instead of the raw state variables.

I can send over my best results later in a new merge request this evening if you're interested. I tried to keep my changes minor relative to Adam's original work such as on only using the "duration" in the reward and not using any state variables such as angle or position.

**Comment by vmoens** [comment]
> @vmoens No worries at all. I did try tweaking various hyperparameters and doing reward shaping to get it to train within 500 to 1000 episodes. As with others, I saw non-monotonic reward history, where the agent would perform well then forget how to solve the problem again. I believe this is to be expected for vanilla DQN learning from pixels instead of the raw state variables.
> 
> 
> 
> I can send over my best results in a new merge request later this evening (GMT -4) if you're interested (still only able to consistently get ~100 timesteps by 1000 episodes, but I haven't tried saving the "best" model to see how it generalizes). I tried to keep my changes minor relative to Adam's original work such as on only using the "duration" in the reward and not using any state variables such as angle or position.

Sure let's keep iterating on this. There are some things I'm trying ok my side too!
Do you know if it used to train and then it didn't or did we always have this performance?

**Comment by SiftingSands** [comment]
> Sure let's keep iterating on this. There are some things I'm trying ok my side too! Do you know if it used to train and then it didn't or did we always have this performance?

Unfortunately, I do not know if it trained successfully in the past. I looked through a few blogs to see if anyone reported it working, but all successful examples involved significant modifications such as using the state variables instead of pixels. I wonder if it's always been somewhat "broken", since there's this issue from 2018 https://github.com/pytorch/tutorials/pull/209 about missing the target network. Just speculating here though...

**Comment by arr00** [comment]
Why isn't this being merged? I followed the current tutorial with no luck but this modified version works perfectly.
